src/dune/link_time_code_gen.ml

```diff
@@ -160,7 +160,7 @@ let build_info_code cctx ~libs ~api_version =
   pr buf "[@@inline never]";
   pr buf "";
   Path.Source.Map.iteri !placeholders ~f:(fun path var ->
-      pr buf "let %s = eval %S" var
+      pr buf "let %s = eval (Sys.opaque_identity %S)" var
```

Comment:
This is not available in 4.02 BTW, so we need to check the compiler version.

Reply:
what should we do on 4.02?

Reply:
I can also try `[@@inlined never]` at the call site

Reply:
If that doesn't work I'd be tempted to try to round-trip the string through printf and scanf.

Reply:
at least on opam there's no 4.02+flambda switch so this might not be a problem. I'll double check

Reply:
Indeed. Using @jberdine 's method of printf+scanf on 4.02 seems good as well